HB/common/utils.elpi

```diff
@@ -252,3 +252,42 @@ w-params_1 X Y :- w-params.then X ignore ignore (p\ t\ triple_1) Y.
 pred purge-id i:term, o:term.
 purge-id T T1 :-
   (pi fresh t v\ copy {{lib:@hb.id lp:t lp:v}} fresh :- !) => copy T T1.
+
+% Extract coercion gref
+pred coercion_gref i:coercion, o:gref.
+coercion_gref (coercion GR _ _ _) GR.
+
+% I could not find this in the API
+% This is extremely slow compared to what it should be.
+pred coq.coercion? i:gref, o:coercion.
+coq.coercion? GR Coe :- std.do! [
+  if-verbose (coq.say "HB: is" GR "a coercion?"),
+  coq.coercion.db Coes,
```

Comment:
It is coq.coercion.db-for, if it is not returning the one you want it means we declare them in the wrong order

Reply:
That's not what the API provides... it needs the source and target class as input, and I want a gref as input...